runtime/commonMain/src/kotlinx/benchmark/BenchmarkReportFormatter.kt

```diff
@@ -0,0 +1,201 @@
+        appendSpace(l - value.length).append(value)
+
+    private fun StringBuilder.appendPaddedAfter(value: String, l: Int): StringBuilder =
+        append(value).appendSpace(l - value.length)
+
+
+    /**
+     * Algorithm:
+     *  1. remove package names, if it is the same for all benchmarks
+     *  2. if not, shorthand same package names
+     *
+     *  (jmh similar logic)
+     */
+    private fun denseBenchmarkNames(src: List<String>): Map<String, String> {
+        if (src.isEmpty()) return emptyMap()
+
+        var first = true
+        var prefixCut = false
+
+        val prefix = src.fold(emptyList<String>()) { prefix, s ->
+            val names = s.split(".")
+            if (first) {
+                first = false
+                names.takeWhile { it.toLowerCase() == it }
+            } else {
+                val common = prefix.zip(names).takeWhile { (p, n) -> p == n && n.toLowerCase() == n }
+                if (prefix.size != common.size) prefixCut = true
+                prefix.take(common.size)
+            }
+        }.map { if (prefixCut) it[0].toString() else "" }
+
+        return src.associateWith { s ->
+            val names = prefix + s.split(".").drop(prefix.size)
+            names.joinToString("") { if (it.isNotEmpty()) "$it." else "" }.removeSuffix(".")
+        }
+    }
+}
+
+private class CsvBenchmarkReportFormatter(val delimiter: String) : BenchmarkReportFormatter() {
+    override fun format(results: Collection<ReportBenchmarkResult>): String = buildString {
+        val allParams = results.flatMap { it.params.keys }.toSet()
+        appendHeader(allParams)
+        results.forEach {
+            appendResult(allParams, it)
+        }
+    }
+
+    private fun StringBuilder.appendHeader(params: Set<String>) {
+        appendEscaped("Benchmark").append(delimiter)
+        appendEscaped("Mode").append(delimiter)
+        appendEscaped("Threads").append(delimiter)
+        appendEscaped("Samples").append(delimiter)
+        appendEscaped("Score").append(delimiter)
+        appendEscaped("Score Error (99.9%)").append(delimiter) //TODO
+        appendEscaped("Unit")
+        params.forEach {
+            append(delimiter)
+            append("Param: $it")
+        }
+        append("\r\n")
+    }
+
+    private fun StringBuilder.appendResult(params: Set<String>, result: ReportBenchmarkResult) {
+        appendEscaped(result.benchmark.name).append(delimiter)
+        appendEscaped(result.config.mode.toText()).append(delimiter)
+        append(1).append(delimiter)
+        append(result.values.size).append(delimiter)
+        append(result.score.format(6, useGrouping = false)).append(delimiter)
+        append(result.error.format(6, useGrouping = false)).append(delimiter)
+        appendEscaped(unitText(result.config.mode, result.config.outputTimeUnit))
+        params.forEach {
+            append(delimiter)
+            result.params[it]?.let { param ->
```

Comment:
Is it possible to get a `null` value here?
If it not, usualy `getValue(it)` is used

Reply:
yes, it's possible, when there are several different benchmark classes in one configuration, with different parameters.

Reply:
Got it, thanks